src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -315,6 +315,15 @@ public Future<String> indexDataverse(Dataverse dataverse, boolean processPaths)
         }
         
         solrInputDocument.addField(SearchFields.SUBTREE, dataversePaths);
+
+        if (dataverse.isReleased()) {
+            // Get all datasets published in this dataverse, including harvested datasets
+            int numberOfPublishedDatasets = dataverseService.findAllDataverseDatasetChildren(dataverse.getId(), true, true).size();
```

Comment:
It might be worth testing this versus a custom query at scale (or, alternatively, to just test this versus the current indexing on a large database). I _think_ this should just result in a proxy list where calling size() won't try to instantiate a real list of dataset objects, but I'm not sure. If it does do extra work compared to just a select count(*) query for the right datasets, it could slow or increase memory requirements significantly.